uefi/src/proto/media/load_file.rs

```diff
@@ -0,0 +1,86 @@
+//! Load file support protocols.
+
+#[cfg(feature = "alloc")]
+use alloc::vec::Vec;
+
+use crate::proto::device_path::{FfiDevicePath, DevicePath};
+use crate::proto::unsafe_protocol;
+use crate::{Result, Status};
+use core::ffi::c_void;
+use core::ptr;
+
+/// The UEFI LoadFile2 protocol.
+///
+/// This protocol has a single method to load a file according to some
+/// device path.
+///
+/// This interface is implemented by many devices, e.g. network and filesystems.
+#[derive(Debug)]
+#[repr(C)]
+#[unsafe_protocol("4006c0c1-fcb3-403e-996d-4a6c8724e06d")]
+pub struct LoadFile2Protocol {
+    load_file: unsafe extern "efiapi" fn(
+        this: &mut LoadFile2Protocol,
+        file_path: *const FfiDevicePath,
+        boot_policy: bool,
+        buffer_size: *mut usize,
+        buffer: *mut c_void,
+    ) -> Status,
+}
+
+impl LoadFile2Protocol {
+    /// Load file addressed by provided device path
+    pub fn load_file(&mut self,
+        file_path: &DevicePath,
+        boot_policy: bool,
```

Comment:
The [spec](https://uefi.org/specs/UEFI/2.10/13_Protocols_Media_Access.html#efi-load-file2-protocol-loadfile) says `boot_policy` should always be false, so let's drop this parameter from the public API here and in the heap version.

(Why the spec writers included a parameter that must always be false is beyond me, I guess they really wanted `LoadFile` and `LoadFile2` to have the same API for some reason.)

Reply:
Whereas, I think for LoadFile, it _should_ always be equal to true. At least, in the source code, boot policy = false in some scenarios will return EFI_INVALID_PARAMETER.

Reply:
Addressed.